crates/proto/src/error.rs

```diff
@@ -56,7 +61,8 @@ macro_rules! trace {
 }
 
 /// An alias for results returned by functions of this crate
-pub(crate) type ProtoResult<T> = ::std::result::Result<T, ProtoError>;
+#[allow(clippy::alloc_instead_of_core)]
```

Comment:
Why is this `allow()` needed when the alias actually uses `core`?

Reply:
Good point, confusion between core::result and core::error on my end